### PR TITLE
Made develop mode autorun on start

### PR DESCRIPTION
**Pull request by notzippy**
- Modified proxy so application is launched on start of proxy, so a new request is not required to start the application.
 - Moved watcher inside harness

### CODE REVIEW

**Comment by brendensoares** [comment]
Is watcher not used within the revel lib?

**Comment by notzippy** [comment]
For watching template files, so it can reload them on a change. The source
files have to be watched from harness.. may be able to do this differently
in the future though

File: harness/harness.go [hunk 1]
```diff
@@ -61,13 +61,16 @@ func (h *Harness) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
 
-	// Flush any change events and rebuild app if necessary.
-	// Render an error page if the rebuild / restart failed.
-	err := watcher.Notify()
-	if err != nil {
-		atomic.CompareAndSwapInt32(&lastRequestHadError, 0, 1)
-		renderError(w, r, err)
-		return
+	// If app did not start when harness was run then trigger the build to capture the error
+	if h.app==nil {
```
Comment (brendensoares) [inline harness/harness.go]: `h.app == nil`

File: harness/harness.go [hunk 2]
```diff
@@ -61,13 +61,16 @@ func (h *Harness) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+		// Flush any change events and rebuild app if necessary.
+		// Render an error page if the rebuild / restart failed.
+		err := h.watcher.Notify()
+		if err != nil {
+			atomic.CompareAndSwapInt32(&lastRequestHadError, 0, 1)
```
Comment (brendensoares) [inline harness/harness.go]: `CompareAndSwapInt32` is kind of obscure, we should add comments to lines referencing this method to communicate our intent which I believe has to do with syncing up the concurrent change to `lastRequestHadError`

File: harness/harness.go [hunk 3]
```diff
@@ -167,8 +170,9 @@ func (h *Harness) Run() {
 		paths = append(paths, gopaths...)
 	}
 	paths = append(paths, revel.CodePaths...)
-	watcher = revel.NewWatcher()
-	watcher.Listen(h, paths...)
+	h.watcher = revel.NewWatcher()
+	h.watcher.Listen(h, paths...)
+	h.watcher.Notify()
```
Comment (brendensoares) [inline harness/harness.go]: Is this the line that causes the build in dev mode?

Comment (notzippy) [inline harness/harness.go]: Yup, the notify triggers the build

Comment (notzippy) [inline harness/harness.go]: `h.watcher.Notify()` is yes

File: harness/harness.go (hunk 2 above)
Comment (notzippy) [inline harness/harness.go]: You are correct, I guess there could be a race condition where the browser requests the favicon immediately following the page request. I don't know if it is crucial to be this way - but it isnt hurting anything by being there

File: harness/harness.go (hunk 1 above)
Comment (notzippy) [inline harness/harness.go]: updated

File: harness/harness.go (hunk 3 above)
Comment (brendensoares) [inline harness/harness.go]: maybe add a comment saying so so the intent is clear?

**Pull request merged**